emission/analysis/userinput/matcher.py

```diff
@@ -63,32 +63,88 @@ def handle_single_most_recent_match(confirmed_trip, ui):
 
 def handle_multi_non_deleted_match(confirmed_trip, ui):
     logging.debug(f"handling user input {ui} for {confirmed_trip}")
-    if "trip_addition" not in confirmed_trip["data"] or \
-        confirmed_trip["data"]["trip_addition"] is None:
-        confirmed_trip["data"]["trip_addition"] = []
+    if "additions" not in confirmed_trip["data"] or \
+        confirmed_trip["data"]["additions"] is None:
+        confirmed_trip["data"]["additions"] = []
     if "status" not in ui.data or ui.data.status == ecwtui.InputStatus.ACTIVE:
-        confirmed_trip["data"]["trip_addition"].append(ui)
+        confirmed_trip["data"]["additions"].append(ui)
     elif ui.data.status == ecwtui.InputStatus.DELETED:
-        after_del_list = [ta for ta in confirmed_trip["data"]["trip_addition"] if ta["match_id"] != ui["match_id"]]
-        confirmed_trip["data"]["trip_addition"] = after_del_list
+        after_del_list = [ta for ta in confirmed_trip["data"]["additions"] if ta["match_id"] != ui["match_id"]]
+        confirmed_trip["data"]["additions"] = after_del_list
     else:
         # TODO: Decide whether to error or to warn here
         logging.warn("Invalid status found in user input %s, moving forward anyway" % ui)
 
+def create_composite_objects(user_id):
+    time_query = epq.get_time_range_for_composite_object_creation(user_id)
+    ts = esta.TimeSeries.get_time_series(user_id)
+    confirmedTrips = esda.get_entries(esda.CONFIRMED_TRIP_KEY, user_id, time_query=time_query)
+    logging.info("Creating composite trips from %d confirmed trips" % len(confirmedTrips))
+    if len(confirmedTrips) == 0:
+        logging.debug("len(confirmedTrips) == 0, early return")
+        return None
+    for ct in confirmedTrips:
+        logging.info("End place type for trip is %s" %  type(ct.data.end_place))
+        end_place = esda.get_entry(esda.CONFIRMED_PLACE_KEY, ct["data"]["confirmed_place"])
+        composite_trip_dict = copy.copy(ct)
+        del composite_trip_dict["_id"]
+        composite_trip_dict["metadata"]["key"] = "analysis/composite_trip"
+        composite_trip_dict["data"]["confirmed_place"] = end_place
+
+        # later we will want to put section & modes in composite_trip as well
+
+        composite_trip_entry = ecwe.Entry(composite_trip_dict)
+        # save the entry
+        ts.insert(composite_trip_entry)
```

Comment:
This is the correct high-level approach. Note, however, that eventually, we do want to put the trajectory in here. But we can defer that until we are done with the current sprint for place and it is tested end to end.

Reply:
Added to to-do section